closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -261,13 +264,36 @@ static void printErrors(void)
     fflush(stderr);
 }
 
+/*
+ * We use a 8 digit map (AABBCCDD) to represent the version of openssl.
+ * AA is the major version,
+ * BB is the minor version,
+ * CC is the revision, and DD for the letter that could be present in any version.
+ * For example, if an openssl version is in this scheme 1.1.1k.
+ * The ossl_ver should be 10101011 ( k is the 11th letter value for DD here... ).
+ *
+*/
+long ver_to_l(const char* astring) {
+    long major = 0;
+    long minor = 0;
+    long revision = 0;
+    long diff = 0;
+    sscanf(astring, "OpenSSL %ld.%ld.%ld", &major, &minor, &revision);
+    char last_char = astring[strlen(astring) - 1];
```

Comment:
The return value of `sscanf()` should not be ignored.

Reply:
Got the return value and added the check for it as well.

Reply:
The return value should be 3 (given we hope to match three values).